setup.py

```diff
@@ -38,12 +43,69 @@
     nightly_build = True
     sys.argv.remove('--nightly_build')
 
+is_manylinux2010 = False
+if environ.get('AUDITWHEEL_PLAT', None) == 'manylinux2010_x86_64':
+    is_manylinux2010 = True
+
+
+class build_ext(_build_ext):
+    def build_extension(self, ext):
+        dest_file = self.get_ext_fullpath(ext.name)
+        logger.info('copying %s -> %s', ext.sources[0], dest_file)
+        copyfile(ext.sources[0], dest_file)
+
+
 try:
     from wheel.bdist_wheel import bdist_wheel as _bdist_wheel
     class bdist_wheel(_bdist_wheel):
         def finalize_options(self):
             _bdist_wheel.finalize_options(self)
-            self.root_is_pure = False
+            if not is_manylinux2010:
+                self.root_is_pure = False
+
+        def _rewrite_ld_preload(self, to_preload):
+            with open('onnxruntime/capi/_ld_preload.py', 'rt') as f:
+                ld_preload = f.read().splitlines()
+            with open('onnxruntime/capi/_ld_preload.py', 'wt') as f:
+                for line in ld_preload:
+                    f.write(line)
+                    f.write('\n')
+                    if 'LD_PRELOAD_BEGIN_MARK' in line:
+                        break
+                if len(to_preload) > 0:
+                    f.write('from ctypes import CDLL, RTLD_GLOBAL\n')
+                    for library in to_preload:
+                        f.write('_{} = CDLL("{}", mode=RTLD_GLOBAL)\n'.format(library.split('.')[0], library))
+
+        def run(self):
+            if is_manylinux2010:
+                source = 'onnxruntime/capi/onnxruntime_pybind11_state.so'
+                dest = 'onnxruntime/capi/onnxruntime_pybind11_state_manylinux2010.so'
+                logger.info('copying %s -> %s', source, dest)
+                copyfile(source, dest)
+                result = subprocess.run(['patchelf', '--print-needed', dest], check=True, stdout=subprocess.PIPE, universal_newlines=True)
+                cuda_dependencies = ['libcublas.so', 'libcudnn.so', 'libcudart.so']
+                to_preload = []
```

Comment:
Could you please give an example of the to_preload list?

Reply:
`['libcublas.so.10.0', 'libcudnn.so.7', 'libcudart.so.10.0']`

Reply:
Why they need be preloaded? I guess they can be automatically loaded when needed.

Reply:
In order for `auditwheel` to create a manylinux2010 compliant wheel, cuda libraries dependencies are removed from the python C extension, thus they need to be preloaded.